x-pack/metricbeat/module/googlecloud/stackdriver/metrics_requester.go

```diff
@@ -0,0 +1,137 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package stackdriver
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/libbeat/logp"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/pkg/errors"
+	"google.golang.org/api/iterator"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+
+	"github.com/elastic/beats/x-pack/metricbeat/module/googlecloud"
+)
+
+func newStackdriverMetricsRequester(c config, window time.Duration, logger *logp.Logger) (*stackdriverMetricsRequester, error) {
+	interval, err := getTimeInterval(window)
+	if err != nil {
+		return nil, errors.Wrap(err, "error trying to get time window")
+	}
+
+	client, err := monitoring.NewMetricClient(context.Background(), c.opt)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating Stackdriver client")
+	}
+
+	return &stackdriverMetricsRequester{
+		config:   c,
+		client:   client,
+		logger:   logger,
+		interval: interval,
+	}, nil
+}
+
+type stackdriverMetricsRequester struct {
+	config config
+
+	client   *monitoring.MetricClient
+	interval *monitoringpb.TimeInterval
+
+	logger *logp.Logger
+}
+
+func (r *stackdriverMetricsRequester) Metric(ctx context.Context, m string) (out []*monitoringpb.TimeSeries) {
+	out = make([]*monitoringpb.TimeSeries, 0)
+
+	req := &monitoringpb.ListTimeSeriesRequest{
+		Name:     "projects/" + r.config.ProjectID,
+		Interval: r.interval,
+		View:     monitoringpb.ListTimeSeriesRequest_FULL,
+		Filter:   fmt.Sprintf(`metric.type="%s"`, m),
```

Comment:
Is it possible to concatenate multiple `metric.type="%s"` with `OR` to request multiple time series at once?

Reply:
Short answer: Nope. Long answer: No but we can do other things like asking for more than one region or asking for metrics "prefixed with" or "labelled with". Everything was described on the proposal, here you have a link to the document in GCP about the filters https://cloud.google.com/monitoring/api/v3/filters The implications, however, are that we might loose or make very complex the possibility to use Stackdriver aggregations in the future. We decided on purpose that we will keep things simple for the beta. Good question